src/rsc/server-richtext.ts

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { StoryblokServerComponent } from './common';
+import type { StoryblokRichTextNode, StoryblokRichTextOptions } from '@storyblok/js';
+import { BlockTypes, richTextResolver } from '@storyblok/js';
+
+export function componentResolver(node: StoryblokRichTextNode<React.ReactElement>): React.ReactElement {
+  const body = node?.attrs?.body;
+  return React.createElement(StoryblokServerComponent, {
+    blok: Array.isArray(body) ? body[0] : undefined,
+    key: node.attrs?.id,
```

Comment:
Risk of an undefined key

```suggestion
    key: node.attrs?.id || `storyblok-component-${Date.now()}`,
```